test/functional/specs/C12411.js

```diff
@@ -0,0 +1,37 @@
+import { t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../helpers/networkLogger";
+import fixtureFactory from "../helpers/fixtureFactory";
+import { orgMainConfigMain } from "../helpers/constants/configParts";
+import configureAlloyInstance from "../helpers/configureAlloyInstance";
+import createResponse from "../../../src/core/createResponse";
+import getResponseBody from "../helpers/networkLogger/getResponseBody";
+
+const networkLogger = createNetworkLogger();
+fixtureFactory({
+  title:
+    "C12411 Response should return URL destinations if turned on in Blackbird",
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+test.meta({
+  ID: "C12411",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+const triggerAlloyEvent = ClientFunction(() =>
```

Comment:
Let's update this function to:

```
const triggerAlloyEvent = ClientFunction(() => {
  return new Promise(resolve => {
    window.alloy("event", { xdm: { key: "value" } }).then(resolve)
  });
});
```

Reply:
This will prevent any race conditions. We will be replacing all those `triggerEvents` with a util very soon.

Reply:
I think you can just return the promise from `window.alloy` directly:

```
const triggerAlloyEvent = ClientFunction(() => {
  return window.alloy("event", { xdm: { key: "value" } });
});
```